### PR TITLE
Fixing #339 & adding persistence mode in xml

### PR DESCRIPTION
**Pull request by AbfrmBlr**

### CODE REVIEW

File: impl/src/test/java/org/ehcache/GettingStarted.java [hunk 1]
```diff
@@ -178,6 +178,29 @@ public void testPersistentDiskCache() {
...
+    PersistentCacheManager persistentCacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+        .with(new PersistenceConfiguration(new File(System.getProperty("java.io.tmpdir") + "/persistent-cache-data")))
+        .withCache("persistent-cache", cacheConfiguration)
+        .build(true);
+
+    Cache<Long, String> cache = persistentCacheManager.getCache("persistent-cache", Long.class, String.class);
+
+    // Comment the following line on subsequent run and see the test pass
+    cache.put(42L, "That's the answer!");
+    assertThat(cache.get(42L), is("That's the answer!"));
```
Comment (lorban) [inline impl/src/test/java/org/ehcache/GettingStarted.java]: The assertion is comparing strings here. While is() actually works (it does ==), it's more correct to use equalTo() (which calls equals()).

File: xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java [hunk 2]
```diff
@@ -156,7 +157,21 @@ private void parseConfiguration()
...
+      if(cacheDefinition.persistenceMode() != null) {
+        CacheConfiguration.PersistenceMode mode = null;
+        switch (cacheDefinition.persistenceMode()) {
+        case CREATE_IF_ABSENT: 
+          mode = CacheConfiguration.PersistenceMode.CREATE_IF_ABSENT ;
+          break;
+        case EXPECT_EXISTS:
+          mode = CacheConfiguration.PersistenceMode.EXPECT_EXISTS;
+          break;
+        default:
```
Comment (lorban) [inline xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java]: I'd prefer to see "case SWAP:" and have the default throw IllegalArgumentException.

File: xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java [hunk 3]
```diff
@@ -292,6 +307,21 @@ public URL getURL() {
...
+    if(cacheTemplate.persistenceMode() != null) {
+      CacheConfiguration.PersistenceMode mode = null;
+      switch (cacheTemplate.persistenceMode()) {
+      case CREATE_IF_ABSENT: 
+        mode = CacheConfiguration.PersistenceMode.CREATE_IF_ABSENT ;
+        break;
+      case EXPECT_EXISTS:
+        mode = CacheConfiguration.PersistenceMode.EXPECT_EXISTS;
+        break;
+      default:
```
Comment (lorban) [inline xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java]: same remark as above

File: xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java (hunk 2 above)
Comment (AbfrmBlr) [inline xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java]: But already xsd takes care of all the validations, so in any case default maps to only one that is SWAP. And SWAP is anyways the default mode of persistence.

Comment (lorban) [inline xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java]: Call me paranoid, but I think an XML config file that doesn't specify the XSD is going to be accepted by the code. Plus if the XSD ever gets out of sync with the code sometime in the future (think of a bug in a future code change) then the bug would be a lot easier to track down.
Or do you think I'm being unreasonably paranoid?

Comment (ljacomet) [inline xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java]: I agree with Ludovic here

**Comment by ljacomet** [comment]
Needs to be reworked - closing this

**Pull request closed without merging**